File: code_samples/elixir/file_operations_reading.ex
# File Operations: Reading
# AI/ML Training Sample

defmodule Reading do
  defstruct data: ""
  
  def new(), do: %Reading{}
  
  def process(%Reading{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Reading{data: data}), do: data
  
  def validate(%Reading{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Reading.new()
updated = Reading.process(instance, "example")
IO.puts("Data: " <> Reading.get_data(updated))
IO.puts("Valid: " <> to_string(Reading.validate(updated)))

File: code_samples/rust/file_operations_reading.rs
//! File Operations: Reading
//! AI/ML Training Sample

pub struct Reading {
    data: String,
}

impl Reading {
    pub fn new() -> Self {
        Self {
            data: String::new(),
        }
    }
    
    pub fn process(&mut self, input: &str) {
        self.data = input.to_string();
    }
    
    pub fn get_data(&self) -> &str {
        &self.data
    }
    
    pub fn validate(&self) -> bool {
        !self.data.is_empty()
    }
}

fn main() {
    let mut instance = Reading::new();
    instance.process("example");
    println!("Data: {}", instance.get_data());
    println!("Valid: {}", instance.validate());
}
